text/utf8/module.f.ts

```diff
@@ -1,104 +1,246 @@
+/**
+ * An unsigned 8-bit integer, represents a single byte.
+ */
+export type U8 = number;
 
-export type Utf8NonEmptyState = Array1<number> | Array2<number> | Array3<number>
+/**
+ * A singed 32-bit integer.
+ */
+export type I32 = number;
 
-export type Utf8State = null | Utf8NonEmptyState
+/**
+ * Represents an unsigend 8-bit type - U8 or the end-of-file indicator.
+ * The U8 represents the byte itself, and null indicates that reading does not return anything else.
+ */
+export type ByteOrEof = U8 | null;
 
-export type U8 = number
+/**
+ * Represents the state of a UTF-8 decoding operation that contains at least one byte.
+ */
+export type Utf8NonEmptyState =
+  | Array1<number>
+  | Array2<number>
+  | Array3<number>;
 
-export type I32 = number
+/**
+ * Represents the state of a UTF-8 decoding operation, which can be either `null` (no state)
+ * or a non-empty state containing one or more bytes.
+ */
+export type Utf8State = null | Utf8NonEmptyState;
 
-const errorMask = 0b1000_0000_0000_0000_0000_0000_0000_0000
+/**
+ * Error mask constant used to represent invalid code points or encoding errors in UTF-8.
+ */
+const errorMask = 0b1000_0000_0000_0000_0000_0000_0000_0000;
 
+/**
+ * Converts a Unicode code point to a sequence of UTF-8 bytes.
+ * @param input The Unicode code point to be converted. Valid range:
+ *   - 0x0000 to 0x007F for 1-byte sequences.
+ *   - 0x0080 to 0x07FF for 2-byte sequences.
+ *   - 0x0800 to 0xFFFF for 3-byte sequences.
+ *   - 0x10000 to 0x10FFFF for 4-byte sequences.
+ * @returns A readonly array of UTF-8 bytes representing the input code point.
+ *   - Returns `[errorMask]` if the input does not match valid UTF-8 encoding rules.
+ */
 const codePointToUtf8 = (input: number): readonly U8[] => {
-    if (input >= 0x0000 && input <= 0x007f) { return [input & 0b01111_1111] }
-    if (input >= 0x0080 && input <= 0x07ff) { return [input >> 6 | 0b1100_0000, input & 0b0011_1111 | 0b1000_0000] }
-    if (input >= 0x0800 && input <= 0xffff) { return [input >> 12 | 0b1110_0000, input >> 6 & 0b0011_1111 | 0b1000_0000, input & 0b0011_1111 | 0b1000_0000] }
-    if (input >= 0x10000 && input <= 0x10ffff) { return [input >> 18 | 0b1111_0000, input >> 12 & 0b0011_1111 | 0b1000_0000, input >> 6 & 0b0011_1111 | 0b1000_0000, input & 0b0011_1111 | 0b1000_0000] }
-    if ((input & errorMask) !== 0) {
-        if ((input & 0b1000_0000_0000_0000) !== 0) { return [input >> 12 & 0b0000_0111 | 0b1111_0000, input >> 6 & 0b0011_1111 | 0b1000_0000, input & 0b0011_1111 | 0b1000_0000] }
-        if ((input & 0b0000_0100_0000_0000) !== 0) { return [input >> 6 & 0b0000_1111 | 0b1110_0000, input & 0b0011_1111 | 0b1000_0000] }
-        if ((input & 0b0000_0010_0000_0000) !== 0) { return [input >> 6 & 0b0000_0111 | 0b1111_0000, input & 0b0011_1111 | 0b1000_0000] }
-        if ((input & 0b0000_0000_1000_0000) !== 0) { return [input & 0b1111_1111] }
+  if (input >= 0x0000 && input <= 0x007f) return [input & 0b01111_1111];
```

Comment:
```suggestion
  if (input >= 0x0000 && input <= 0x007f) { return [input & 0b01111_1111] }
```
We use `{` after `if (...)`